ProcessOut/Classes/FingerPrintWebViewSchemeHandler.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  FingerPrintWebViewSchemeHandler.swift
+//
+//  Created by Jeremy Lejoux on 27/08/2019.
+//
+
+import Foundation
+import WebKit
+
+@available(iOS 11.0, *) // When not available we fallback to default fingerprinting values and continue the authorization flow
+public class FingerPrintWebViewSchemeHandler: NSObject, WKURLSchemeHandler {
+    
+    private var completion: ((String?, String?, ProcessOutException?) -> Void)? = nil
+    
+    public init(completion: @escaping (String?, String?, ProcessOutException?) -> Void) {
+        self.completion = completion
+    }
+    
+    public func webView(_ webView: WKWebView, start urlSchemeTask: WKURLSchemeTask) {
+        DispatchQueue.global().async {
+            if let url = urlSchemeTask.request.url {
+                var invoice = ""
+                var token = ""
+                if let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: true)?.queryItems {
+                    for queryParams in queryItems {
+                        if queryParams.name == "invoice_id", let value = queryParams.value {
+                            invoice = value
+                        } else if queryParams.name == "token", let value = queryParams.value {
+                            token = value
+                        }
+                    }
+                }
+                
+                if invoice == "" || token == "" {
+                    self.completion!(nil, nil, ProcessOutException.InternalError)
```

Comment:
You might want to add a friendly message here

Reply:
Unfortunately, the iOS SDK only support error messages coming from the API. I can however log something in the dev console

Reply:
I'm not sure I understand what you mean by "only support error messages coming from the API". Does it mean we can't return errors with messages when not coming from the ProcessOut API?

Reply:
Yep. For now we only have a few different error "types" being:
```swift
public enum ProcessOutException: Error {
    case NetworkError
    case MissingProjectId
    case BadRequest(errorMessage: String, errorCode: String)
    case InternalError
    case GenericError(error: Error)
}
```

I don't want to add custom messages to the `BadRequest` case as I want to keep `errorCode` coming from the API/documentation.

We could add a custom case but merchants will need to update their integration to support it.